lms/djangoapps/discussion_api/views.py

```diff
@@ -192,6 +203,18 @@ def partial_update(self, request, thread_id):
         """
         return Response(update_thread(request, thread_id, request.DATA))
 
+    def destroy(self, request, thread_id):
+        """
+        Implements the DELETE method for instance endpoint as described in the
+        class docstring
+
+        Returns:
+
+            Response with status code 204 upon success.
+
+        """
+        return Response(delete_thread(request, thread_id), status=204)
```

Comment:
I think this will actually make the response contain something. I think you have to do

```
delete_thread(request, thread_id)
return Response(status=204)
```